test/metric.jl

```diff
@@ -235,15 +239,13 @@ end
         g2 = DefaultBaseManifoldMetric()
         MM2 = MetricManifold(M,g2)
 
-        @test is_decorator_manifold(MM) == Val{true}()
-        @test is_decorator_manifold(MM2) == Val{true}()
-        @test is_decorator_manifold(M) == Val{false}()
-        @test is_default_metric(MM) == is_default_metric(base_manifold(MM),metric(MM))
-        @test is_default_metric(MM2) == is_default_metric(base_manifold(MM2),metric(MM2))
-        @test is_default_metric(MM2) == Val{true}()
+        @test (@inferred is_default_metric(MM2))
```

Comment:
And there should be an `@inferred` test for the `_dispatch` function.

Reply:
I am happy to have understood macros today, I haven't understood `@inferred` yet – but yes there should be.

Reply:
Cool, macros are rarely useful but sometimes can save a lot of repetitive work.
`@inferred` just makes sure that the compiler can figure out the exact type of the object returned by a function (which is very important for performance in some places). We don't need it everywhere but it's good to have it for key functions.